platform/state/platform_state.proto

```diff
@@ -0,0 +1,206 @@
+     */
+    proto.Timestamp freeze_time = 6;
+
+    /**
+     * A timestamp for the last time a freeze was performed.<br/>
+     * If not set, there has never been a freeze.
+     */
+    proto.Timestamp last_frozen_time = 7;
+
+    /**
+     * A consensus node semantic version.<br/>
+     * The software version that enabled birth round mode.
+     * <p>
+     * This SHALL be unset if birth round migration has not yet happened.<br/>
+     * If birth round migration is complete, this SHALL be the _first_ software
+     * version that enabled birth round mode.
+     */
+    proto.SemanticVersion first_version_in_birth_round_mode = 8;
+
+    /**
+     * A consensus round.<br/>
+     * The last round before the birth round mode was enabled.
+     * <p>
+     * This SHALL be `MAX_UNSIGNED` if birth round mode has not yet been enabled.
+     */
+    uint64 last_round_before_birth_round_mode = 9;
+
+    // Fields below are to be deprecated in the foreseeable future.
+
+    /**
+     * A running event hash.<br/>
+     * This is computed by the consensus event stream.
+     * <p>
+     * This will be _removed_ and the field number reserved once the consensus
+     * event stream is retired.
+     */
+     bytes legacy_running_event_hash = 10000 [deprecated = true];
+
+    /**
+     * A consensus generation.<br/>
+     * The lowest judge generation before birth round mode was enabled.
+     * <p>
+     * This SHALL be `MAX_UNSIGNED` if birth round mode has not yet been enabled.
+     */
+    uint64 lowest_judge_generation_before_birth_round_mode = 10001 [deprecated = true];
+}
+
+
+/**
+ * A consensus snapshot.<br/>
+ * This is a snapshot of the consensus state for a particular round.
+ *
+ * This message SHALL record consensus data necessary for restart
+ * and reconnect.
+ */
+message ConsensusSnapshot {
+    /**
+     * A consensus round.<br/>
+     * The round number of this snapshot.
+     */
+    uint64 round = 1;
+
+    /**
+     * A list of SHA-384 hash values.<br/>
+     * The hashes of all judges for this round.
+     * <p>
+     * This list SHALL be ordered by creator ID.<br/>
+     * This list MUST be deterministically ordered.
+     */
+    repeated bytes judge_hashes = 2;
+
+    /**
+     * A list of minimum judge information entries.<br/>
+     * These are "minimum ancient" entries for non-ancient rounds.
+     */
+    repeated MinimumJudgeInfo minimum_judge_info_list = 3;
+
+    /**
+     * A single consensus number.<br/>
+     * The consensus order of the next event to reach consensus.
+     */
+    uint64 next_consensus_number = 4;
+
+    /**
+     * A "consensus" timestamp.<br/>
+     * The consensus timestamp of this snapshot.
+     * <p>
+     * This SHALL be a consensus value and MAY NOT correspond to an actual
+     * "wall clock" timestamp.<br/>
+     * Consensus Timestamps SHALL always increase.
+     */
+    proto.Timestamp consensus_timestamp = 5;
+}
+
+/**
+ * Minimum ancient threshold information for round judges.<br/>
```

Comment:
I'm struggling to understand what this comment describes.

Does this structure represent the minimum ancient threshold for a _single_ judge? Or multiple judges at once, as implied by the plural usage of `judges` in this description?

Reply:
I updated the description